spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -381,7 +387,20 @@ def reset_streams():
     print("git origin: {}\n".format(__get_git_origin(path_to_git_folder=path_data)))
 
     # Find subjects and process inclusion/exclusions
-    subject_dirs = [f for f in os.listdir(path_data) if f.startswith(arguments.subject_prefix)]
+    subject_dirs = []
+    subject_flat_dirs = [f for f in os.listdir(path_data) if f.startswith(arguments.subject_prefix)]
+    for isub in subject_flat_dirs:
+        # Only consider folders
+        if os.path.isdir(os.path.join(path_data, isub)):
+            session_dirs = [f for f in os.listdir(os.path.join(path_data, isub)) if f.startswith('ses-')]
+            if not session_dirs:
+                # There is no session folder, so we consider only sub- directory: sub-XX
+                subject_dirs.append(isub)
+            else:
+                # There is a session folder, so we concatenate: sub-XX/ses-YY
+                session_dirs.sort()
+                for isess in session_dirs:
+                    subject_dirs.append(os.path.join(isub, isess))
```

Comment:
This sort of parsing behavior seems like the sort of thing that the BIDS team would have already implemented themselves. A quick search yields the [PyBIDS](https://github.com/bids-standard/pybids) project, and this page looks especially relevant: https://bids-standard.github.io/pybids/layout/index.html

Do you know if any of NeuroPoly's projects have tried using PyBIDS? We might want to consider adopting this to avoid having to re-invent the wheel, and not risk missing edge-cases/weird folder naming behaviors, etc.

Reply:
excellent suggestion-- yes, i thought about this and yes, some folks in the lab are using PyBIDS. Unfortunately I don't have the time to investigate, but maybe @alexfoias can help here. I suggest we open an issue so we can merge this PR asap (i need it for a project that i need to deliver on very quickly)

Reply:
> (i need it for a project that i need to deliver on very quickly)

This is important context! Just to clarify: Do you need this to be in a release, or is having it on `master` fine?

Reply:
thank you for asking-- master is fine

Reply:
I'll open a separate issue for this suggestion and leave it for now, then. :slightly_smiling_face:

Reply:
We are using pybids in ivadomed to index a bids dataset and create a dataframe here: https://github.com/ivadomed/ivadomed/blob/1d81b32c61891e4b5b4f2ffbbb7923e367111dc9/ivadomed/loader/utils.py#L639